backend/x86_proc.mli

```diff
@@ -89,4 +89,5 @@ val use_plt : bool
 
 (** Support for plumbing a binary code emitter *)
 
+val internal_assembler : (asm_program -> string -> unit) option ref
```

Comment:
With `internal_assembler` exposed as a `ref` here, there is no need for `register_internal_assembler` any more. Do you really need it to be a `ref`?

Reply:
In the JIT, I need to be able to get its current value, set it to our own internal_assembler function and once we're done, set it back to is original value (be it `None` or `Some f`).

Exposing the `ref` seemed simpler than adding the range of functions I would need to implement the above but in the end I'm happy with whatever solution you prefer.
Note that I left the `register_internal_assembler` function in order not to break the compiler API.

I haven't submitted the patches to the upstream compiler yet. Once that's done we should make sure the flambda2 compiler and upstream have the same toplevel API. In the meantime please let me know what suits you best and I'll update the PR!

Reply:
Makes sense if you need to set it back to `None`.